frontend/components/user/settings/UserSettingsEmail.vue

```diff
@@ -0,0 +1,148 @@
+<script lang="ts" setup>
+import { object as yupObject } from 'yup'
+import { useForm } from 'vee-validate'
+import { API_PATH } from '~/types/customFetch'
+
+const { t: $t } = useI18n()
+const { fetch } = useCustomFetch()
+const toast = useBcToast()
+
+const { handleSubmit, errors, defineField } = useForm({
+  validationSchema: yupObject({
+    password: passwordValidation($t),
+    newEmail: emailValidation($t),
+    confirmEmail: confirmEmailValidation($t, 'newEmail')
+  })
+})
+
+const [password, passwordAttrs] = defineField('password')
+const [newEmail, newEmailAttrs] = defineField('newEmail')
+const [confirmEmail, confirmEmailAttrs] = defineField('confirmEmail')
+
+const buttonsDisabled = defineModel<boolean | undefined>({ required: true })
+
+const onSubmit = handleSubmit(async (values) => {
+  if (!canSubmit.value) {
+    return
+  }
+
+  buttonsDisabled.value = true
+  try {
+    await fetch(API_PATH.USER_CHANGE_EMAIL, {
+      body: {
+        password: values.password,
+        email: values.newEmail
+      }
+    })
```

Comment:
maybe a success toast would be nice ... the user does not get any feedback right now (if success)

Reply:
Good idea, added. Furthermore the form now gets reset on success.

Reply:
get an A+ for this one dough